app/service/planning_svc.py

```diff
@@ -78,7 +78,7 @@ def __init__(self, data_svc, utility_svc):
             variable_facts = []
             for fact in facts:
                 if fact['property'] == v:
-                    variable_facts.append((fact['property'], fact['value'], fact['score'], fact['id']))
+                    variable_facts.append((fact['property'], fact['value'], fact['score'], fact['id'], fact['set_id'], fact['link_id']))
```

Comment:
really good PR idea here & nice clean approach to your code. two things i think it could use:
1) at line 81, since we're starting to pull in most of the properties of a fact here, do you think we should just include the full fact dict() instead of pulling out specific properties? I think that's probably better, as we won't have to modify this again if we decide we want something else that's on the fact.
2) at line 98; is there a special meaning behind needing the combo_set/link_id variables to equal 1 or are we just doing a check to ensure it is not empty? if the latter, can we just do 
``
if combo_set_id and combo_link_id:
    ....
```

Reply:
1. Yes, we could pull the whole dict(). I'll go ahead and do that.
2. We check if the set length is 1 (and only 1) for the following reason. Consider Fact A and Fact B. If A.set_id = B.set_id, then the length of combo_set_id will be 1. If A.set_id != B.set_id, then combo_set_id will be 2.